### PR TITLE
Sampler schema

### PR DESCRIPTION
**Pull request by jeremydvoss**
Sampler id is basically the stand in for OTEL_TRACES_SAMPLER's use of entry points. It includes entry points under the opentelemetry_traces_sampler entry point group.

Extra properties are true so that those samplers or sampler factories can take additional arguments.

### CODE REVIEW

**Comment by tsloughter** [comment]
I don't think we should restrict the types of things to strings to match environment variables. Unless the schema is being limited to just taking an arg, like `OTEL_TRACE_SAMPLER_ARG`, and not defining a schema for each builtin sampler type.

File: schema/sampler.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "additionalProperties": true,
+    "properties": {
+        "sampler_id": {
+            "type": "string"
+        },
+        "argument": {
+            "type": "string"
+        }
+    }
+}
```
Comment (jack-berg) [inline schema/sampler.json:15]: You'll need to extend [kitchen-sink-example.yaml](https://github.com/open-telemetry/opentelemetry-configuration/blob/main/kitchen-sink-example.yaml) to show example usage.

Also, sampler will be defined as a part of [trace_provider](https://github.com/open-telemetry/opentelemetry-configuration/blob/main/schema/tracer_provider.json), so you'll need to extend that schema to include the appropriate fields.

**Review by jack-berg — COMMENTED** [review]
We've already done some thinking about sampler configuration. Check out [this example configuration](https://github.com/MrAlias/otel-schema/blob/main/config.yaml#L178-L229) for the types of things that need to be configurable and our first attempt at structuring that info.

File: schema/sampler.json (hunk 1 above)
Comment (jeremydvoss) [reply schema/sampler.json:12]: I believe I have addressed this now.

**Comment by jeremydvoss** [comment]
> I don't think we should restrict the types of things to strings to match environment variables. Unless the schema is being limited to just taking an arg, like `OTEL_TRACE_SAMPLER_ARG`, and not defining a schema for each builtin sampler type.

With better understanding from the last sig, I no longer need sampler arg. the samplers arguments will be represented by the additional properties.

File: kitchen-sink-example.yaml [hunk 2]
```diff
@@ -112,6 +112,26 @@ propagators: [tracecontext, baggage, b3, b3multi, jaeger, xray, ottrace]
 
 # Configure tracer provider.
 tracer_provider:
+  sampler:
+    # Choose sampler id. Must be entry point name of sampler
+    sampler_id: parent_based
```
Comment (jeremydvoss) [inline kitchen-sink-example.yaml:117]: I decided sampler_id is clearer than "type" which we used in our weekly sig @jack-berg Thoughts?

**Comment by jack-berg** [comment]
Superseded by #24.

**Pull request closed without merging**